src/legacy/core_plugins/data/public/search/search_bar/components/search_bar.test.tsx

```diff
@@ -0,0 +1,204 @@
+
+jest.mock('../../../filter/filter_bar', () => {
+  return {
+    FilterBar: () => <div className="filterBar"></div>,
+  };
+});
+
+jest.mock('../../../query/query_bar', () => {
+  return {
+    QueryBar: () => <div className="queryBar"></div>,
+  };
+});
+
+const noop = jest.fn();
+
+const createMockWebStorage = () => ({
+  clear: jest.fn(),
+  getItem: jest.fn(),
+  key: jest.fn(),
+  removeItem: jest.fn(),
+  setItem: jest.fn(),
+  length: 0,
+});
+
+const createMockStorage = () => ({
+  store: createMockWebStorage(),
+  get: jest.fn(),
+  set: jest.fn(),
+  remove: jest.fn(),
+  clear: jest.fn(),
+});
+
+const mockIndexPattern = {
+  id: '1234',
+  title: 'logstash-*',
+  fields: [
+    {
+      name: 'response',
+      type: 'number',
+      esTypes: ['integer'],
+      aggregatable: true,
+      filterable: true,
+      searchable: true,
+    },
+  ],
+};
+
+const kqlQuery = {
+  query: 'response:200',
+  language: 'kuery',
+};
+
+describe('SearchBar', () => {
+  const SEARCH_BAR_ROOT = '.globalQueryBar';
+  const FILTER_BAR = '.filterBar';
+  const QUERY_BAR = '.queryBar';
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Should render query bar when no options provided (in reality - timepicker)', () => {
+    const component = mountWithIntl(
+      <SearchBar.WrappedComponent
+        appName={'test'}
+        indexPatterns={[mockIndexPattern]}
+        intl={null as any}
+      />
+    );
+
+    expect(component.find(SEARCH_BAR_ROOT).length).toBe(1);
+    expect(component.find(FILTER_BAR).length).toBe(0);
+    expect(component.find(QUERY_BAR).length).toBe(1);
+  });
+
+  it('Should render empty when timepicker disabled and no options provided', () => {
+    const component = mountWithIntl(
+      <SearchBar.WrappedComponent
+        appName={'test'}
+        indexPatterns={[mockIndexPattern]}
+        intl={null as any}
+        showDatePicker={false}
+      />
+    );
+
+    expect(component.find(SEARCH_BAR_ROOT).length).toBe(1);
+    expect(component.find(FILTER_BAR).length).toBe(0);
+    expect(component.find(QUERY_BAR).length).toBe(0);
+  });
+
+  it('Should render filter bar, when required fields are provided', () => {
+    const component = mountWithIntl(
+      <SearchBar.WrappedComponent
+        appName={'test'}
+        indexPatterns={[mockIndexPattern]}
+        intl={null as any}
+        filters={[]}
+        onFiltersUpdated={noop}
+        showDatePicker={false}
+      />
+    );
+
+    expect(component.find(SEARCH_BAR_ROOT).length).toBe(1);
+    expect(component.find(FILTER_BAR).length).toBe(1);
+    expect(component.find(QUERY_BAR).length).toBe(0);
+  });
+
+  it('Should NOT render filter bar, if disabled', () => {
+    const component = mountWithIntl(
+      <SearchBar.WrappedComponent
+        appName={'test'}
+        indexPatterns={[mockIndexPattern]}
+        intl={null as any}
+        showFilterBar={false}
+        filters={[]}
+        onFiltersUpdated={noop}
+        showDatePicker={false}
```

Comment:
Date picker will show if not disabled explicitly, because it can work without any options.
I know it's not optimal, and I intend to address this in a separate issue that will re-think the API of search bar \ filter bar \ query bar compoents.